xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -770,8 +770,22 @@ bool CGUIMediaWindow::Update(const std::string &strDirectory, bool updateFilterP
   }
 
   if (m_vecItems->GetLabel().empty())
+  {
     m_vecItems->SetLabel(CUtil::GetTitleFromPath(m_vecItems->GetPath(), true));
 
+    // Removable sources
```

Comment:
Yeah something like that if we always want the label be set if a removable path match.